site/src/pages/release-notes/1.27.0.mdx

```diff
@@ -0,0 +1,228 @@
+- message processing using <type://WebSocketServiceBuilder#aggregateContinuation(boolean)>. #5357
+  ```java
+  WebSocketService
+    .builder()
+    // Aggregate continuation frames into a single frame.
+    .aggregateContinuation(true)
+    .build(); 
+  ```
+- **Active Connection Monitoring**: You can track the number of active connections with 
+  `armeria.client.connections#value{state="active"}` for your clients when using 
+  <type://ConnectionPoolListener#metricCollecting(MeterRegistry)> #4277 #5288
+  ```java
+  ClientFactory clientFactory =
+    ClientFactory
+      .builder()
+      .connectionPoolListener(ConnectionPoolListener.metricCollecting(meterRegistry))
+      .build();
+  WebClient
+    .builder("https://my-backend.com")
+    .factory(clientFactory)
+    .build();
+  ```
+- **Graceful Executor Shutdown**: You can now gracefully terminate a custom `ExecutorService` in sync with the
+- <type://Server>'s shutdown procedure using <type://ServerListenerBuilder#shutdownWhenStopped(ExecutorService,Duration)>
+  and <type://ServerListenerBuilder#shutdownWhenStopped(ExecutorService,Duration)> #5031 #5087
+  ```java
+  ServerListener
+    .builder()
+    // Shutdown the executor service when the server is stopping.
+    .shutdownWhenStopping(preStopExecutorService)
+    // Shutdown the executor service when the server is stopped.
+    .shutdownWhenStopped(postStopExecutorService)
+    .build()
+  ```
+- **Session Protocol in RoutingContext**: You can now acquire <type://SessionProtocol> directly from
+  <type://RoutingContext#sessionProtocol()> for enhanced routing and <type://EchangeType> decisions.
+- You can now easily mask request and response headers with ... #5175 #5188
+  - TODO(ikhoon): TBU
+- **Eureka Health Check Customization**: You can override the default health check URL when registering with 
+  Eureka. #5369
+  ```java
+  EurekaUpdatingListener
+    .builder()
+    // Override the default health check path provided by HealthCheckService.
+    .healthCheckUrlPath("/healthcheck")
+    .build();
+  ```
+
+## 📈 Improvements
+
+- Annotated services now support <type://HttpResult> wrapped in RxJava objects #5380 #5386
+- You can now create <type://HttpFile> with 
+  [Spring Boot nested JAR URL](https://docs.spring.io/spring-boot/docs/current/reference/html/executable-jar.html#appendix.executable-jar.jarfile-class.compatibility). #5390 #5391
+- <type://WebsocketService> is restructured as an interface, allowing for custom implementations. #5368
+- Introduce <type://PreTlsHandshakeException> to better report exceptions occurring before a TLS handshake
+  initiates. #5344
+- Error messages for gRPC client stub generation have been clarified. #5387
+
+## 🛠️ Bug fixes
+
+- <type://ClientRequestContext#whenResponseCancelling()> and <type://ClientRequestContext#whenResponseCancelled()>
+  is now correctly completed even if the  `CancellationScheduler` has not been initialized #5212
```

Comment:
```suggestion
  are now correctly completed even if the <type://CancellationScheduler> has not been initialized. #5212
```

Reply:
`CancellationScheduler` is an internal API so the sentence has been rewritten with different words.